internal/k8sutil/k8sutil.go

```diff
@@ -150,21 +153,56 @@ func QuoteAttestationDelete(ctx context.Context, c client.Client, instanceName s
 		},
 	}
 
-	err := c.Delete(ctx, sgxAttestation)
-	if err != nil && errors.IsNotFound(err) {
-		return nil
+	if err := client.IgnoreNotFound(UnsetFinalizer(ctx, c, sgxAttestation)); err != nil {
+		return fmt.Errorf("failed unset finalizer for '%s/%s': %v", ns, instanceName, err)
 	}
-	return err
+
+	return client.IgnoreNotFound(c.Delete(ctx, sgxAttestation))
 }
 
-// Converts signer name to valid Kubernetes object name
-//  Ex:- intel.com/tcs -> tcs.intel.com
-//       tcsissuer.tcs.intel.com/sgx-ca1 -> sgx-ca1.tcsissuer.intel.tcs.com
-func SignerNameToResourceName(signerName string) string {
+// Converts signer name to valid Kubernetes object name and nanespace
+//  Ex:- intel.com/tcs -> tcs.intel.com, ""
+///      tcsissuer.tcs.intel.com/sandbox.sgx-ca -> sgx-ca.tcs.intel.com, sandbox
+//       tcsclusterissuer.tcs.intel.com/sgx-ca1 -> sgx-ca1.tcsclusterissuer.intel.tcs.com, ""
+func SignerNameToResourceNameAndNamespace(signerName string) (string, string) {
 	slices := strings.SplitN(signerName, "/", 2)
 	if len(slices) == 2 {
-		return slices[1] + "." + slices[0]
+		nameParts := strings.SplitN(slices[1], ".", 2)
+		if len(nameParts) == 2 {
+			return nameParts[1] + "." + slices[0], nameParts[0]
+		}
+		return slices[1] + "." + slices[0], ""
 	}
 
-	return slices[0]
+	return slices[0], ""
+}
+
+func UnsetFinalizer(ctx context.Context, c client.Client, obj client.Object) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Duration(2*time.Minute))
```

Comment:
Why the 2 min timeout?

Reply:
No specific reason :) Do you suggest reducing the timeout?

Reply:
No need, just curious about the choice.

Reply:
I removed the timeout from this function, and linked the internal calls to passed context `ctx`.